File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int N = nums.length;
        
        int xor1 = 0, xor2 = 0;

        for (int i = 0; i < N - 1; i++) {
            xor2 = xor2 ^ nums[i]; 
            xor1 = xor1 ^ (i + 1); 
        }
        xor1 = xor1 ^ N;

        return (xor1 ^ xor2);
    }
}

File: 0268-missing-number/0268-missing-number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        summ = 0

        for num in nums:
            summ += num

        arr_sum = (n * (n+1)) // 2

        return arr_sum - summ
        